test/dict.jl

```diff
@@ -374,6 +376,32 @@ let
     Base.show(Base.IOContext(IOBuffer(), :limit => true), a)
 end
 
+let
+    a = ObjectIdDict()
+    a[1] = a
+    a[a] = 2
```

Comment:
Maybe something like this:
```jl
@test length(a) == 2
@test 1 in keys(a)
@test a in keys(a)
@test a[1] === a
@test a[a] === 2
```
In other words check that `a` is the object we thing we've constructed: an associative collection with exactly two keys, which are `1` and `a`, mapped to `a` and `2`, respectively.